packages/typescript-estree/src/semantic-or-syntactic-errors.ts

```diff
@@ -56,43 +56,45 @@ function whitelistSupportedDiagnostics(
-      case 1176: // ts 3.2 "Interface declaration cannot have 'implements' clause."
-      case 1190: // ts 3.2 "The variable declaration of a 'for...of' statement cannot have an initializer."
-      case 1200: // ts 3.2 "Line terminator not permitted before arrow."
-      case 1206: // ts 3.2 "Decorators are not valid here."
-      case 1211: // ts 3.2 "A class declaration without the 'default' modifier must have a name."
-      case 1242: // ts 3.2 "'abstract' modifier can only appear on a class, method, or property declaration."
-      case 1246: // ts 3.2 "An interface property cannot have an initializer."
-      case 1255: // ts 3.2 "A definite assignment assertion '!' is not permitted in this context."
-      case 1308: // ts 3.2 "'await' expression is only allowed within an async function."
-      case 2364: // ts 3.2 "The left-hand side of an assignment expression must be a variable or a property access."
-      case 2369: // ts 3.2 "A parameter property is only allowed in a constructor implementation."
-      case 2462: // ts 3.2 "A rest element must be last in a destructuring pattern."
-      case 8017: // ts 3.2 "Octal literal types must use ES2015 syntax. Use the syntax '{0}'."
-      case 17012: // ts 3.2 "'{0}' is not a valid meta-property for keyword '{1}'. Did you mean '{2}'?"
-      case 17013: // ts 3.2 "Meta-property '{0}' is only allowed in the body of a function declaration, function expression, or constructor."
+      case 1013: // "A rest parameter or binding pattern may not have a trailing comma."
+      case 1014: // "A rest parameter must be last in a parameter list."
+      case 1044: // "'{0}' modifier cannot appear on a module or namespace element."
+      case 1045: // "A '{0}' modifier cannot be used with an interface declaration."
+      case 1048: // "A rest parameter cannot have an initializer."
+      case 1049: // "A 'set' accessor must have exactly one parameter."
+      case 1070: // "'{0}' modifier cannot appear on a type member."
+      case 1071: // "'{0}' modifier cannot appear on an index signature."
+      case 1085: // "Octal literals are not available when targeting ECMAScript 5 and higher. Use the syntax '{0}'."
+      case 1090: // "'{0}' modifier cannot appear on a parameter."
+      case 1096: // "An index signature must have exactly one parameter."
+      case 1097: // "'{0}' list cannot be empty."
+      case 1098: // "Type parameter list cannot be empty."
+      case 1099: // "Type argument list cannot be empty."
+      case 1117: // "An object literal cannot have multiple properties with the same name in strict mode."
+      case 1121: // "Octal literals are not allowed in strict mode."
+      case 1123: //  "Variable declaration list cannot be empty."
+      case 1141: // "String literal expected."
+      case 1162: // "An object member cannot be declared optional."
+      case 1164: // "Computed property names are not allowed in enums."
```

Comment:
new line

Reply:
if computed properties in enum names are not valid than you don't have to handle this case in eslint rules.

Reply:
looks like this line is untested,
code like this should be ok
```ts
enum E {
  [e] = 1
}
```

https://github.com/microsoft/TypeScript/blob/master/tests/baselines/reference/parserES5ComputedPropertyName6.errors.txt

Reply:
This code here is all pretty well unused. I only made this change because I thought we automatically threw parser errors if it was listed here.

But then I learned that it is only ever hit if someone passes the undocumented `errorOnTypeScriptSyntacticAndSemanticIssues` option.